hack/deploy-karmada-agent.sh

```diff
@@ -60,8 +60,10 @@ kubectl config use-context "${MEMBER_CLUSTER_NAME}"
 # load image if member cluster created by kind (if kubeconfig clusters' name contains kind-xxx)
 if kubectl config get-clusters | grep "kind-${MEMBER_CLUSTER_NAME}"
 then
-  export REGISTRY="swr.ap-southeast-1.myhuaweicloud.com/karmada"
+  # make agent image
   export VERSION="latest"
+  export REGISTRY="swr.ap-southeast-1.myhuaweicloud.com/karmada"
+  make image-karmada-agent --directory="${REPO_ROOT}"
```

Comment:
It seems like we don't need to build the image of karmada agent if you set imagePullPolicy as Always.

Reply:
Yeap, it does. I tuned it just now. such as @RainbowMango 's said. There are two cases, one is for member cluster created by kind, in this case, it needs to build the latest image of karmada agent and set the image pulling policy to `IfNotPresent` to make sure load the local build image. Another case is for the standalone member cluster, it does not need to build the image and set the image pulling policy to `Always` to pull the latest image.

Reply:
I make a simple judgment in https://github.com/lfbear/karmada/blob/issues-434/hack/deploy-karmada-agent.sh#L62. Is there the word 'kind-' in the cluster name